### PR TITLE
sys-fs/zfs: install compatibility.d

### PR DESCRIPTION
**Pull request by implr**
Added upstream in https://github.com/openzfs/zfs/pull/11468, didn't work on Gentoo as we remove the entirety of `/usr/share/zfs/`.
I admit the brace/glob expansion is a bit ugly.

### CODE REVIEW

File: sys-fs/zfs/zfs-2.1.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+}
+
+src_install() {
+	default
+
+	gen_usr_ldscript -a nvpair uutil zfsbootenv zfs zfs_core zpool
+
+	use pam && { rm -rv "${ED}/unwanted_files" || die ; }
+
+	use test-suite || { rm -r "${ED}/usr/share/zfs/"{test-runner,zfs-tests,runfiles,*sh} || die ; }
```
Comment (gyakovlev) [inline sys-fs/zfs/zfs-2.1.1-r1.ebuild:224]: `find "${ED}/usr/share/zfs" -mindepth 1 -maxdepth 1 -not -name compatibility.d -exec rm -r '{}' \;`

this should work better =)

Comment (thesamesam) [reply sys-fs/zfs/zfs-2.1.1-r1.ebuild:224]: Problem is, the PR version makes it easier to not miss things in future.. hm

Comment (implr) [reply sys-fs/zfs/zfs-2.1.1-r1.ebuild:224]: Yeah I considered something like the find, but then when they add more files there we'll drop them again and not notice.
I'm not good with autotools; would it be feasible to instead patch out/disable installing the test files at an earlier stage?

Comment (gyakovlev) [reply sys-fs/zfs/zfs-2.1.1-r1.ebuild:224]:
```diff
diff --git a/sys-fs/zfs/zfs-2.1.1.ebuild b/sys-fs/zfs/zfs-2.1.1.ebuild
index cf525674e3ab..f141ae27d79e 100644
--- a/sys-fs/zfs/zfs-2.1.1.ebuild
+++ b/sys-fs/zfs/zfs-2.1.1.ebuild
@@ -154,6 +154,13 @@ src_prepare() {
        default
        libsoversion_check
 
+       # remove those before build, and before autoreconf
+       if ! use test-suite; then
+               rm -rf tests || die
+               sed -i '/tests\/.*Makefile/d' configure.ac || die
+               sed -i '/^SUBDIRS\ +=/ s/tests//' Makefile.am || die
+       fi
+
        # Run unconditionally (bug #792627)
        eautoreconf
 
@@ -221,8 +228,6 @@ src_install() {
 
        use pam && { rm -rv "${ED}/unwanted_files" || die ; }
 
-       use test-suite || { rm -r "${ED}/usr/share/zfs" || die ; }
-
        find "${ED}" -name '*.la' -delete || die
 
        dobashcomp contrib/bash_completion.d/zfs
```

this is pre-build deletion version messing with autofiles.
tbh looks fragile, and leaves some extra dangling files too.
I think the PR version is fine in the long run. so let's merge it.

Comment (thesamesam) [reply sys-fs/zfs/zfs-2.1.1-r1.ebuild:224]: I worry about *sh matching something it shouldn't, but at some point, we can try upstream a proper way to not install these. Let's add a comment and merge the PR version?

**Comment by gyakovlev** [comment]
merged, restored riscv keyword (ebuild was keyworded meanwhile) and moved quotes a bit, just cosmetic.
thanks for great first contribution to gentoo ;-) really useful.

@sam feel free to add comment to ebuild ofc if you want.

**Comment by gyakovlev** [comment]
and yeah, I agree that some kind of configure switch upstream would be nice. installing those files unconditionally does not make a lot of sense for end users.

**Comment by implr** [comment]
Thanks. I'll bring that up upstream if I find some time later.

**Pull request closed without merging**